src/categorical_algebra/CSetDataStructures.jl

```diff
@@ -299,18 +300,23 @@ function incident_body(s::SchemaDesc, idxed::AbstractDict{Symbol,Bool},
 end
 
 @generated function _incident(acs::StructACSet{S,Ts,Idxed,UniqueIdxed},
-                              part, ::Type{Val{f}}; copy::Bool=false) where
-  {S,Ts,Idxed,UniqueIdxed,f}
-  incident_body(SchemaDesc(S),pairs(Idxed),pairs(UniqueIdxed),f)
+                              part, ::Type{Val{f}}, ::Type{Val{copy}}) where
+  {S,Ts,Idxed,UniqueIdxed,f,copy}
+  incident_body(SchemaDesc(S),pairs(Idxed),pairs(UniqueIdxed),f,copy)
 end
 
 # Mutators
 ##########
 
-@inline ACSetInterface.add_parts!(acs::StructACSet, ob::Symbol, n::Int) = _add_parts!(acs, Val{ob}, n)
+@inline add_parts_with_indices!(acs::StructACSet, ob::Symbol, n::Int, index_sizes::NamedTuple) =
```

Comment:
Is this function now part of the public interface? If so, it should have tests and documentation.